File: TravelEase/src/main/java/com/viniorsi/TravelEase/Service/Email/EmailConsumer.java
package com.viniorsi.TravelEase.Service.Email;

import com.viniorsi.TravelEase.Domain.Email.EmailMessage;
import com.viniorsi.TravelEase.Infra.RabbiMq.RabbitMQConfig;
import org.springframework.amqp.rabbit.annotation.RabbitListener;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

@Service
public class EmailConsumer {

    @Autowired
    private EmailService emailService;

    @RabbitListener(queues = RabbitMQConfig.QUEUE_NAME)
    public void receiveEmailMessage(EmailMessage emailMessage) {
        emailService.enviarEmailTexto(
                emailMessage.getTo(),
                emailMessage.getSubject(),
                emailMessage.getBody(),
                emailMessage.getAttachments()
        );
    }

}

File: APP TravelEasy/EmailSmsSender/src/main/kotlin/com/br/TravelEase/EmailSmsSender/Listener/EmailConsumer.kt
package com.br.TravelEase.EmailSmsSender.Listener

import com.br.TravelEase.EmailSmsSender.Config.RabbitMQConfig
import org.springframework.amqp.rabbit.annotation.RabbitListener
import org.springframework.stereotype.Service

@Service
class EmailConsumer(
    private val emailService: EmailService
) {


    @RabbitListener(queues = [RabbitMQConfig.QUEUE_NAME])
    fun receiveEmailMessage(emailMessage: EmailMessage) {
        emailService.enviarEmailTexto(
            emailMessage.to,
            emailMessage.subject,
            emailMessage.body,
            emailMessage.attachments as MutableList<String>
        );
    }

}